Unable to compile with Mono on custom build

**Goost and Godot version:**
latest `gd3` commit, and custom Godot build based on latest `3.x` commit


**OS/platform/device including version:**
<!-- Specify GPU model, drivers, and the backend (GLES2, GLES3, Vulkan) if graphics-related. -->
Arch Linux
```
dotnet --list-sdks
5.0.403 [/usr/share/dotnet/sdk]
6.0.102 [/usr/share/dotnet/sdk]
```


**Issue description:**
I want to build Goost against my custom Godot build with Mono support.  Compiling with mono glue enabled fails with the following errors:
```
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/MeshLibrary.cs(71,25): error CS0117: 'NativeCalls' non contiene una definizione per 'godot_icall_2_647' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/MeshLibrary.cs(95,25): error CS0117: 'NativeCalls' non contiene una definizione per 'godot_icall_2_647' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/MeshLibrary.cs(108,25): error CS0117: 'NativeCalls' non contiene una definizione per 'godot_icall_2_648' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/MeshInstance.cs(158,32): error CS0117: 'NativeCalls' non contiene una definizione per 'godot_icall_0_476' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/MeshLibrary.cs(144,86): error CS1503: Argomento 3: non è possibile convertire da 'int' a 'System.IntPtr' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/MeshLibrary.cs(156,25): error CS0117: 'NativeCalls' non contiene una definizione per 'godot_icall_1_650' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/MeshInstance.cs(194,32): error CS0117: 'NativeCalls' non contiene una definizione per 'godot_icall_1_634' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/MeshInstance.cs(206,32): error CS0117: 'NativeCalls' non contiene una definizione per 'godot_icall_1_634' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/MeshDataTool.cs(67,39): error CS0117: 'NativeCalls' non contiene una definizione per 'godot_icall_2_638' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/MeshDataTool.cs(79,39): error CS0117: 'NativeCalls' non contiene una definizione per 'godot_icall_1_639' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]

... # and a  lot more of the same kind
```


**Steps to reproduce:**
Clone Goost and symlink custom godot source tree inside Goost local repository. Use scons as described in the Goost and Godot documentation for building from source: i compile with mono enabled and no glue, then i generate the glue, and then i recompile with glue.
```
scons p=x11 target=release_debug tools=yes bits=64 module_mono_enabled=yes mono_glue=no -j8
godot/bin/godot.x11.opt.tools.64.goost.mono --generate-mono-glue godot/modules/mono/glue
scons p=x11 target=release_debug tools=yes bits=64 module_mono_enabled=yes mono_glue=yes-j8
```

**Is this issue reproducible in official Godot builds?**
No.

**Minimal reproduction project:**
N/A

The changes in my PR fix the aforementioned issue, but there is an additional problem with these errors being thrown during the final step of a Mono build
```
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/Debug2D.cs(188,73): error CS1503: Argomento 2: non è possibile convertire da 'double' a 'float' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/Debug2D.cs(188,77): error CS1503: Argomento 3: non è possibile convertire da 'double' a 'float' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/Debug2D.cs(216,73): error CS1503: Argomento 2: non è possibile convertire da 'double' a 'float' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/Debug2D.cs(216,77): error CS1503: Argomento 3: non è possibile convertire da 'double' a 'float' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/Debug2D.cs(231,73): error CS1503: Argomento 2: non è possibile convertire da 'double' a 'float' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/Debug2D.cs(231,77): error CS1503: Argomento 3: non è possibile convertire da 'double' a 'float' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/Debug2D.cs(245,73): error CS1503: Argomento 2: non è possibile convertire da 'double' a 'float' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/Debug2D.cs(245,77): error CS1503: Argomento 3: non è possibile convertire da 'double' a 'float' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/Debug2D.cs(263,73): error CS1503: Argomento 2: non è possibile convertire da 'double' a 'float' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/Debug2D.cs(263,77): error CS1503: Argomento 3: non è possibile convertire da 'double' a 'float' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/Debug2D.cs(279,73): error CS1503: Argomento 2: non è possibile convertire da 'double' a 'float' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/Debug2D.cs(279,77): error CS1503: Argomento 3: non è possibile convertire da 'double' a 'float' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/Debug2D.cs(295,73): error CS1503: Argomento 2: non è possibile convertire da 'double' a 'float' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/Generated/GodotObjects/Debug2D.cs(295,77): error CS1503: Argomento 3: non è possibile convertire da 'double' a 'float' [/home/glitchedcode/repos/godot/modules/mono/glue/GodotSharp/GodotSharp/GodotSharp.csproj]
```

Which are caused by the way Godot's code generation algorithm generates the glue, passing double constants (the default in C#) to Color's constructor which accepts floats. This can be worked around by adding the following constructor to the Color class in `godot/modules/mono/glue/GodotSharp/GodotSharp/Core/Color.cs`:

```
/// <summary>
/// Constructs a <see cref="Color"/> from RGBA values, typically on the range of 0 to 1.
/// </summary>
/// <param name="r">The color's red component, typically on the range of 0 to 1.</param>
/// <param name="g">The color's green component, typically on the range of 0 to 1.</param>
/// <param name="b">The color's blue component, typically on the range of 0 to 1.</param>
/// <param name="a">The color's alpha (transparency) value, typically on the range of 0 to 1. Default: 1.</param>
public Color(double r, double g, double b, double a = 1.0f)
{
    this.r = (float)r;
    this.g = (float)g;
    this.b = (float)b;
    this.a = (float)a;
}
```
But if you're particularly pedantic about the semantics of floating-point maths this is not optimal. I'm going to open an issue to Godot's repo and eventually a PR to get this sorted out.
On the bright side, Godot and Goost with Mono support now gets compiled just fine.

--- a/doc/MidiFile.xml
+++ b/doc/MidiFile.xml
@@ -23,7 +23,7 @@
 		<member name="data" type="PoolByteArray" setter="set_data" getter="get_data" default="PoolByteArray(  )">
 			A stream of bytes which represents MIDI data. This data is not meant to be modified, as MIDI files are imported automatically.
 		</member>
-		<member name="format" type="int" setter="set_format" getter="get_format" default="0">
+		<member name="format" type="Format" setter="set_format" getter="get_format" default="0">
 			Represents the internal format of the [MidiFile] data. This is set automatically at import time.
 		</member>
 	</members>

--- a/scene/audio/midi_player.h
+++ b/scene/audio/midi_player.h
@@ -51,12 +51,12 @@ public:
 		FORMAT_SF2,
 	};
 	
-	int format = FORMAT_MIDI; // Default.
+	Format format = FORMAT_MIDI; // Default.
 
 	Error load(const String fileName);
 	
 	void set_format(Format p_format) { format = p_format; }
-	int get_format() const { return format; }
+	Format get_format() const { return format; }
 
 	void set_data(const PoolVector<uint8_t> &p_data);
 	PoolVector<uint8_t> get_data() const;

